pkgs/tools/package-management/nix/common.nix

```diff
@@ -163,6 +164,20 @@ self = stdenv.mkDerivation {
     aws-sdk-cpp
   ] ++ lib.optional (atLeast218 && stdenv.hostPlatform.isDarwin) [
     darwin.apple_sdk.libs.sandbox
+  ] ++ lib.optional (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isStatic) [
+    # libarchive requires a iconv.pc file to be present
+    (pkgs.runCommand "iconv-pkgconfig" {} ''
+      mkdir -p $out/lib/pkgconfig
+      cat > $out/lib/pkgconfig/iconv.pc <<EOF
+      prefix=${iconv}
+
+      Name: iconv
+      Description: GNU iconv
+      Version: ${lib.getVersion iconv}
+      Libs: -liconv
+      EOF
+    '')
+    iconv
```

Comment:
Are you sure you want this? This is the `iconv(1)` binary. `libiconv` is already provided by the base SDK.